modular_chomp/code/modules/projectiles/ammo_refactor/ammo_magazine_override.dm

```diff
@@ -0,0 +1,283 @@
+//Replace old magazine with new
+/obj/item/ammo_magazine/New(loc)
+	var/replacement_type = magazine_overrides[type]
+	if(replacement_type)
+		qdel(src)
+		return new replacement_type(loc)
+	. = ..()
+
+
+//Override attackby to allow magazine to magazine transfer
+/obj/item/ammo_magazine/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	if(istype(W, /obj/item/ammo_casing))
+		var/obj/item/ammo_casing/C = W
+		if(C.caliber != caliber)
+			to_chat(user, "<span class='warning'>[C] does not fit into [src].</span>")
+			return
+		if(stored_ammo.len >= max_ammo)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+			return
+		user.remove_from_mob(C)
+		C.forceMove(src)
+		stored_ammo.Add(C)
+		update_icon()
+	if(istype(W, /obj/item/ammo_magazine))
+		var/obj/item/ammo_magazine/L = W
+		if(L.caliber != caliber)
+			to_chat(user, "<span class='warning'>The ammo in [L] does not fit into [src].</span>")
+			return
+		if(!L.stored_ammo.len)
+			to_chat(user, "<span class='warning'>There's no more ammo [L]!</span>")
+			return
+		if(stored_ammo.len >= max_ammo)
+			to_chat(user, "<span class='warning'>[src] is full!</span>")
+			return
+		while(L.stored_ammo.len && stored_ammo.len < max_ammo)
+			var/obj/item/ammo_casing/AC = L.stored_ammo[1] //select the next casing.
+			L.stored_ammo -= AC //Remove this casing from loaded list of the clip.
+			AC.forceMove(src)
+			stored_ammo.Insert(1, AC) //add it to the head of our magazine's list
+		L.update_icon()
+	else
+		return //Don't play sound if nothing actually happens xD
+	playsound(src, 'sound/weapons/flipblade.ogg', 50, 1)
+	update_icon()
+
+//big boy list
+var/global/list/magazine_overrides = list(
```

Comment:
So, if I understand what I am reading correctly, you're not actually removing anything, but you're making some code here to basically be like "if this magazine type comes into existence, make it something else"

which hopefully means if I go ahead and do my own gun PR to try to make it immersive _but still not insane bloat of too many ammo types_ then all I have to do is get rid of this.

I still don't like these changes because I don't like the style of making everything generic, but at least it's easy to get rid of. I'm really more concerned at this point that once these changes get added, it'll be hard to convince people to go back again even if they agree the new changes aren't really ideal. It'll be Polaris "we already changed so much and you're no longer allowed to fuck with it" all over again.